module-2/backend/week_one.md

```diff
@@ -5,30 +5,99 @@ Fork this respository. Answer the questions to the best of your ability. Try to
+    Add => @count = 1 above erb:index 
+    
+8. In the same code block, how would I pass a local variable `name` with a value of `Mr. Ed` to the view? 
+    Add => name = "Mr. Ed"
+9. What's the purpose of ERB? 
+    To allow ruby and html/css to interact together 
+10. Why do I need a development AND test database? 
+    Development is just set/moving things in and around 
+    Test is creating instances that we don't want to stick around after we have tested a thing. 
 11. What's responsive design?
-12. What is CRUD and why is it important?
-13. What does HTTP stand for? 
-14. What are the two ways to interpolate Ruby in an ERB view template? What's the difference between these two ways?
-15. What's an ORM?
+    When you resize a browser window and the site respondse in a way that still makes sense to the user viewing the new size
+12. What is CRUD and why is it important? 
+    Create 
+    Read 
+    Update
+    Delete 
+    
+    These are important because it explains all of the functionality of a website. There isn't anything more that needs to be 
+    done besides C/R/U/D
+13. What does HTTP stand for?  
+    Hyper
+    Text 
+    Transfer
+    Protocol
+14. What are the two ways to interpolate Ruby in an ERB view template? What's the difference between these two ways? 
+    adding <%= @name %> or #{@name}
+15. What's an ORM? 
+    Object 
+    Relational
+    Mapping
 16. What's the most commonly used ORM in ruby (Sinatra & Rails)?
-17. Let's say we have an application with restaurants. There are seven verb + path combinations necessary to provide full CRUD functionality for our restaurant application. List each of the seven combinations, and explain what each is for.
-18. What's a migration? 
-19. When you create a migration, does it automatically modify your database?
+    Rails 
+17. Let's say we have an application with restaurants. There are seven verb + path combinations necessary to provide full CRUD functionality for our restaurant application. List each of the seven combinations, and explain what each is for. 
+    1. /restaurants  
+        Get 
+        - gets :index and displays
+    2.  /restaurants/new  
+        Get 
+        - gets :new and displays, user can enter in form
+    3. /restaurants/new  
+        Post 
+        - once user has selected Submit, the :new form is read and posted to :index 
+    4. /restaurants/:id  
+        Get 
+        - gets :show and displays selected resturant
+    5. /restaurants/:id/edit 
+        Get  
+        - gets :edit and displays the edit form to user where they can enter in or change information
+    6. /restaurants/:id/edit 
+        Post  
+        - once Submmit is selected, form is read and posted to :index
+    7. /restaurants/:id
+        Delete 
+        - takes the selected restuarant out of the list of created items and redirects to :index
+    
+    
+18. What's a migration?  
+      Any change or update made in a SQL database

```

Comment:
I would think of the migrations as the **instructions** to create/update a database.